rust/src/delta_datafusion.rs

```diff
@@ -486,22 +479,6 @@ impl ExecutionPlan for DeltaScan {
         partition: usize,
         context: Arc<TaskContext>,
     ) -> DataFusionResult<SendableRecordBatchStream> {
-        let df_url = ListingTableUrl::parse(self.url.as_str())?;
-        let storage = context
-            .runtime_env()
-            .object_store_registry
-            .get_by_url(df_url);
-        let mut table = DeltaTableBuilder::from_uri(&self.url);
-        if let Ok(storage) = storage {
-            // When running in ballista, the store will be deserialized and re-created
-            // When testing with a MemoryStore, it will already be present and we should re-use it
-            table = table.with_storage_backend(
-                storage,
-                Url::parse(&self.url).map_err(|err| DataFusionError::Internal(err.to_string()))?,
-            );
-        }
-        let table = table.build()?;
-        register_store(&table, context.runtime_env());
```

Comment:
Hmm, I never worked seriously with ballista, so hard for me to guess. Maybe @avantgardnerio, who worked  on the ballista integration can help us out to answer this :)?

Reply:
I'm sure it was necessary at the time. I don't know if the code has been changed since then.

I was waiting for the stars to align for the correct transitive dependencies to not conflict in order to make a Ballista PR, but sadly by the time they did I was refocused on other things.

Apologies about the lack of a test.

Reply:
Got it, no problem.

I've now extended my previous test to also exercise `DeltaScan` serde, along with the accompanying object store registration logic, which I thought would be sufficiently ergonomic for balllista or other such cases. Namely, `DeltaScan` struct keeps the table uri, and the idea is that when someone deserializes a plan they should construct the relevant `DeltaObjectStore` and register it with the current Datafusion state. For this purpose I've also made `DeltaObjectStore::object_store_url` public globally.

Please take a look if that makes sense.